tap_integration_tests/src/lib.rs

```diff
@@ -0,0 +1,633 @@
+use rand::rngs::StdRng;
+use rand::{Rng, SeedableRng};
+use rstest::*;
+use std::collections::{HashMap, HashSet};
+use std::convert::TryInto;
+use std::iter::FromIterator;
+use std::net::SocketAddr;
+use std::net::TcpListener;
+use std::str::FromStr;
+use std::sync::{Arc, RwLock};
+use tap_aggregator::server as agg_server;
+use tap_core::eip_712_signed_message::EIP712SignedMessage;
+use tap_core::tap_receipt::Receipt;
+use tap_core::{
+    adapters::{
+        collateral_adapter_mock::CollateralAdapterMock,
+        rav_storage_adapter_mock::RAVStorageAdapterMock,
+        receipt_checks_adapter_mock::ReceiptChecksAdapterMock,
+        receipt_storage_adapter_mock::ReceiptStorageAdapterMock,
+    },
+    tap_receipt::ReceiptCheck,
+};
+use tokio::join;
+
+pub mod server;
+
+#[fixture]
+fn keys() -> (LocalWallet, Address) {
+    let wallet: LocalWallet = MnemonicBuilder::<English>::default()
+     .phrase("abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about")
+     .build()
+     .unwrap();
+    let address = wallet.address();
+    (wallet, address)
+}
+
+#[fixture]
+fn wrong_keys() -> (LocalWallet, Address) {
+    let wallet: LocalWallet = MnemonicBuilder::<English>::default()
+        .phrase("devote force reopen galaxy humor virtual hobby chief grit nothing bag pulse")
+        .build()
+        .unwrap();
+    let address = wallet.address();
+    (wallet, address)
+}
+
+#[fixture]
+fn allocation_ids() -> Vec<Address> {
+    vec![
+        Address::from_str("0xabababababababababababababababababababab").unwrap(),
+        Address::from_str("0xdeaddeaddeaddeaddeaddeaddeaddeaddeaddead").unwrap(),
+    ]
+}
+
+#[fixture]
+fn http_request_size_limit() -> u32 {
+    100 * 1024
+}
+
+#[fixture]
+fn http_response_size_limit() -> u32 {
+    100 * 1024
+}
+
+#[fixture]
+fn http_max_concurrent_connections() -> u32 {
+    2
+}
+
+#[fixture]
+fn num_queries() -> usize {
+    16
+}
+
+#[fixture]
+fn collateral_adapter() -> CollateralAdapterMock {
+    CollateralAdapterMock::new(Arc::new(RwLock::new(HashMap::new())))
+}
+
+#[fixture]
+fn receipt_storage_adapter() -> ReceiptStorageAdapterMock {
+    ReceiptStorageAdapterMock::new(Arc::new(RwLock::new(HashMap::new())))
+}
+
+#[fixture]
+fn query_price() -> Vec<u128> {
+    let seed: Vec<u8> = (0..32u8).collect(); // A seed of your choice
+    let mut rng: StdRng = SeedableRng::from_seed(seed.try_into().unwrap());
+    let mut v = Vec::new();
+
+    for _ in 0..num_queries() {
+        v.push(rng.gen::<u128>() % 100);
+    }
+    v
+}
+
+#[fixture]
+fn receipt_checks_adapter(query_price: Vec<u128>) -> ReceiptChecksAdapterMock {
+    // Setup receipt storage
+    let receipt_storage = Arc::new(RwLock::new(HashMap::new()));
+
+    let query_prices = query_price;
+    // Setup query appraisals
+    let query_appraisals = (0..num_queries() as u64)
+        .zip(query_prices)
+        .into_iter()
+        .map(|(id, price)| (id, price))
+        .collect::<HashMap<_, _>>();
```

Comment:
```suggestion
    // Setup query appraisals
    let query_appraisals = (0..query_price.len() as u64)
        .zip(query_prices)
        .into_iter()
        .collect::<HashMap<_, _>>();
```

Reply:
It might be more clear to just use the enumerate function here:

```suggestion
    let query_appraisals: HashMap<_, _> = query_price.iter().enumerate().collect();
```